packages/protocol/src/mdns/MdnsScanner.ts

```diff
@@ -1319,6 +1319,9 @@ export class MdnsScanner implements Scanner {
         } else if (result.T === 1) {
             // Value 1 is reserved and should be handled as 0 according to Matter spec
             result.T = 0; // TCP not supported
+        } else if (result.T !== 2 && result.T !== 4 && result.T !== 6) {
```

Comment:
I'd write `!(result.T & 6)`

Reply:
I think the best is to simply clear bit 0 always, then yes that all should be the same

Reply:
so likely 

```!((result.T & ~1) & 6) ```

Reply:
see commit